### DISCUSSION TITLE
Changing: Walk, Run, Sprint

### DISCUSSION THREAD

**Discussion by ItaloLemosGasparo** [Sixze/ALS-Refactored#279] [Q&A]
Hello! I'm neewbie with EU and still learning. I'm looking for a way to make my character only run while Left Shift is pressed and sprint while Lef Alt is pressed. But I'm not finding where can I do that

**Comment by ameaninglessname** [comment]
see `AAlsCharacterExample::Input_OnSprint` and `AAlsCharacterExample::Input_OnWalk`

**Reply by ItaloLemosGasparo** [reply]
But in which file?

**Reply by assmikDuDuDuJ** [reply]
maybe just change the key in >IMC_Als_Default ?

**Reply by ItaloLemosGasparo** [reply]
My problem is more like "Where is the Input actions functions" like Where can i change when I press Shift to run and when release the character back to walk

**Reply by assmikDuDuDuJ** [reply]
I don't get it, if you need a function, the answer was already > in the C++ Code> AlsCharacterExample.cpp

**Reply by ItaloLemosGasparo** [reply]
![image](https://user-images.githubusercontent.com/71532764/231909067-ab7ffb4f-2cfb-475c-9bef-9f0a74747254.png)

![image](https://user-images.githubusercontent.com/71532764/231908978-fd050c93-92ab-4f30-83ce-7cba340b31cd.png)

https://youtu.be/fnvx47klDjI

I found in the original ALS "InputGraph". Is there something like that in Refactured?
(Pics and video from original version of ALS)

**Reply by assmikDuDuDuJ** [reply]
EnhancedInputAction events  "IA_Als_Sprint"\Move\ect. 
 You can find functions like Desired Gait in the "Als Character" Category

**Comment by iJoanx** [comment]
Hello, sorry to resurrect this discussion. I was wondering how to do the same thing. That is, have two states, Walking and Running. When Walking, the IA_Als_Sprint would make me run, and when stopped go back to Walking. When Running, the IA_Als Sprint would make me sprint, and when stopped go back to Running. What would be the easiest way to achieve that?

**Reply by ameaninglessname** [reply]
> when stopped go back to Running

You mean stop the input(button not pressed)?

I think you actually need a `IA_GoToNextGait`: which does `walk` -> `run` -> `sprint`, and there is another `OnMovementStopped` event which does `sprint` -> `run`

which
